ProcessMaker/processmaker
# Import/ Export: Update Import Process Main Page

## Issue & Reproduction Steps
Ticket: [FOUR-6737](https://processmaker.atlassian.net/browse/FOUR-6737)

The main import process page needs to be updated to match the new design specified in the Import/Export [PRD](https://processmaker.atlassian.net/wiki/spaces/PM4/pages/2719481999/PRD+PM4+Export+Import#New-Designer-View.).

## Solution
- Create a shared DraggableFileUpload component to be used on the import process page.
- Create a new `import/validation` endpoint to validate the uploaded file.


**Before**
![Screen Shot 2022-09-19 at 12 42 51 PM](https://user-images.githubusercontent.com/52755494/191101931-e5fba945-d5aa-49f6-a59d-5ee31e6f7cc3.png)


**After**

![Screen Shot 2022-09-19 at 12 26 43 PM](https://user-images.githubusercontent.com/52755494/191101866-84432885-6302-4c87-b01b-1b5f62f04100.png)


## How to Test

Test Case A: PRD Styling
1. Ensure styling matches [PRD](https://processmaker.atlassian.net/wiki/spaces/PM4/pages/2719481999/PRD+PM4+Export+Import#New-Designer-View.).

Test Case B: File Validation

1. Select 'Import' from the `Designer > Process` page
2. Drag and drop or Click 'Select file from computer' 
3. Select an invalid file [collection_export.json.txt](https://github.com/ProcessMaker/processmaker/files/9601916/collection_export.json.txt) (remove .txt)

Expected behavior: 
An 'invalid format' error should appear. The import button is disabled. Another file can be uploaded.

Test Case C: Import Valid File

1. Select 'Import' from the `Designer > Process` page
2. Drag and drop or Click 'Select file from computer' 
3. Select a valid file 
[ABE.json.txt](https://github.com/ProcessMaker/processmaker/files/9601931/ABE.json.txt)
 (remove .txt)
4. Update assignments and select 'Save'

Expected behavior: 
Validation passes. The file imports and you are redirected to the assignments page. The imported process is listed under the Designer > Processes page.

## Related Tickets & Packages
- [Epic Ticket](https://processmaker.atlassian.net/browse/FOUR-6570)
- [Import / Export PRD](https://processmaker.atlassian.net/wiki/spaces/PM4/pages/2719481999/PRD+PM4+Export+Import#New-Designer-View.)

## Code Review Checklist
